src/main/interface/smartaudio_protocol.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "platform.h"
+#include "common/crc.h"
+#include "interface/smartaudio_protocol.h"
+
+#define SMARTAUDIO_SYNC_BYTE                0xAA
+#define SMARTAUDIO_HEADER_BYTE              0x55
+#define SMARTAUDIO_GET_PITMODE_FREQ_BYTE    (1 << 14)
+#define SMARTAUDIO_SET_PITMODE_FREQ_BYTE    (1 << 15)
+#define SMARTAUDIO_FREQUENCY_MASK           0x1FFF
+
+#define SMARTAUDIO_CMD_GET_SETTINGS         0x03
+#define SMARTAUDIO_CMD_SET_POWER            0x05
+#define SMARTAUDIO_CMD_SET_CHANNEL          0x07
+#define SMARTAUDIO_CMD_SET_FREQUENCY        0x09
+#define SMARTAUDIO_CMD_SET_MODE             0x0B
+
+#define SMARTAUDIO_RSP_GET_SETTINGS_V1      SMARTAUDIO_CMD_GET_SETTINGS >> 1
+#define SMARTAUDIO_RSP_GET_SETTINGS_V2      (SMARTAUDIO_CMD_GET_SETTINGS >> 1) | 0x08
+#define SMARTAUDIO_RSP_SET_POWER            SMARTAUDIO_CMD_SET_POWER >> 1
+#define SMARTAUDIO_RSP_SET_CHANNEL          SMARTAUDIO_CMD_SET_CHANNEL >> 1
+#define SMARTAUDIO_RSP_SET_FREQUENCY        SMARTAUDIO_CMD_SET_FREQUENCY >> 1
+#define SMARTAUDIO_RSP_SET_MODE             SMARTAUDIO_CMD_SET_MODE >> 1
+
+#define SMARTAUDIO_BANDCHAN_TO_INDEX(band, channel) ((band) * (uint8_t)8 + (channel))
```

Comment:
Result can still promote to `int`, `(uint8_t)` has no effect

Reply:
Such encoding is usually done in form of `band << 3 | channel` or `band << 3 + channel` rather than multiplication.

Reply:
I pulled this from the current smartaudio implementation.  I'll adjust it.